src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -240,15 +240,7 @@ bool ReadBufferFromRemoteFSGather::readImpl()
     if (!result)
         result = current_buf->next();
 
-    if (blobs_to_read.size() == 1)
-    {
-        file_offset_of_buffer_end = current_buf->getFileOffsetOfBufferEnd();
-    }
-    else
-    {
-        /// For log family engines there are multiple s3 files for the same clickhouse file
-        file_offset_of_buffer_end += current_buf->available();
-    }
+    file_offset_of_buffer_end = current_buf->getFileOffsetOfBufferEnd();
```

Comment:
Why after this change it works correctly if in this case `file_offset_of_buffer_end` will equal file offset of buffer end of current buffer, which is one of the multiple files in s3, but in fact here `file_offset_of_buffer_end` needs to be an accumulated offset among all files as this class acts like a combinator for multiple files in s3 which refer to a single file in ch.

Reply:
>Why after this change it works correctly if in this case file_offset_of_buffer_end will equal file offset of buffer end of current buffer, which is one of the multiple files in s3, but in fact here file_offset_of_buffer_end needs to be an accumulated offset among all files as this class acts like a combinator for multiple files in s3 which refer to a single file in ch.

I guess not enough tests.
But I guess you are right, instead of adjusting `file_offset_of_buffer_end` assertion should be fixed.